### PR TITLE
For #14636 - Update preference section headers to use correct color, …

### PR DESCRIPTION
**Pull request by amkcpu**
...spacing, and rulers.

@eliserichards for review

- Create new attributes for colors
- Switch all preference section headers over to the `preference_cat_style` layout
- Delete unused `preference_category_main_style` to make sure only the right one is used
- Remove attributes suppressing rulers in some settings screens

____
**Note:** https://github.com/mozilla-mobile/fenix/issues/14636#issuecomment-695868874 specifies that in the add-ons management page, the headers should be adjusted too. While it is possible to set the header text color here: https://github.com/mozilla-mobile/fenix/blob/943455658bf90df9c7139512790e065fc7292e0d/app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt#L123, it is currently not possible to set the `allowDividerAbove` attribute. This would need AC work.
I think it would make sense to make it possible to set a header style in `AddonsManagerAdapter` so one could just point to the XML. I think that would be preferable to extending the `Style` data class to be able to set `allowDividerAbove`. What do you think?
____

### Screenshots
Light theme | Dark theme
------------- | -------------
 | 
 | 
 | 

### Pull Request checklist

- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
According to this: https://github.com/mozilla-mobile/fenix/issues/12533 preference section headers on the main screen have a different keyline. You should not delete `preference_category_main_style `.

**Comment by eliserichards** [comment]
bors try

**Review by eliserichards — APPROVED** [review]
This is awesome! Thank you so much for the thorough description and the screenshots! As soon as CI runs & passes we can get this merged 🎉

**Comment by eliserichards** [comment]
bors r=eliserichards

**Comment by eliserichards** [comment]
bors r=eliserichards

**Comment by eliserichards** [comment]
bors r=eliserichards

**Comment by eliserichards** [comment]
bors try

**Comment by pocmo** [comment]
bors try-
bors r-

**Comment by pocmo** [comment]
bors try

**Comment by eliserichards** [comment]
bors try

**Comment by eliserichards** [comment]
bors r=eliserichards

**Comment by eliserichards** [comment]
bors retry

**Comment by pocmo** [comment]
```
[vcs 2020-10-05T20:18:24.763Z] executing ['git', 'clone', 'https://github.com/mozilla-mobile/fenix', '/builds/worker/checkouts/src']
[vcs 2020-10-05T20:18:24.765Z] Cloning into '/builds/worker/checkouts/src'...
[vcs 2020-10-05T20:18:27.674Z] executing ['git', 'fetch', 'https://github.com/mozilla-mobile/fenix', 'refs/heads/staging']
[vcs 2020-10-05T20:18:27.810Z] From https://github.com/mozilla-mobile/fenix
[vcs 2020-10-05T20:18:27.810Z]  * branch                staging    -> FETCH_HEAD
[vcs 2020-10-05T20:18:27.818Z] executing ['git', 'checkout', '-f', '15a38a6f20170a81e53ebcc5229f73badb645d72']
[vcs 2020-10-05T20:18:27.821Z] fatal: reference is not a tree: 15a38a6f20170a81e53ebcc5229f73badb645d72
[taskcluster 2020-10-05 20:18:28.088Z] === Task Finished ===
[taskcluster 2020-10-05 20:18:28.181Z] Unsuccessful task run with exit code: 128 completed in 34.716 seconds
```

weeeeeird. Let's try one more time.

bors retry

**Comment by pocmo** [comment]
This time it ran on Firebase, but some UI tests failed: deleteAllHistoryTest, deleteHistoryItemTest, verifyAboutFirefoxPreview. Since we are only changing the style, I assume that this should not affect the tests and this may be an intermittent failure? Let's try one more time.

bors retry

**Comment by pocmo** [comment]
This time only one failed: verifyAboutFirefoxPreview.

It looks like we need to really take some time and get rid of those intermittent failures.

bors retry

**Comment by pocmo** [comment]
Failed: testDownloadNotification, verifyAboutFirefoxPreview

**Comment by amkcpu** [comment]
@pocmo is there anything I can do to help here?

**Comment by gabrielluong** [comment]
@apbitner @brampitoyo throwing a ux-feedback to the 2 of you! Thanks!

**Comment by apbitner** [comment]
I'll let @brampitoyo review as he provided the specs

**Comment by amkcpu** [comment]
@liuche I merged the remote changes, this should be fine now.

File: app/src/main/res/values/colors.xml
Comment (gabrielluong) [inline app/src/main/res/values/colors.xml:63]: We should have @color variables for these. I think #592ACB maps out to @color/photonViolet70, but also best to do a search.

File: app/src/main/res/values/colors.xml
Comment (gabrielluong) [inline app/src/main/res/values/colors.xml:133]: Can you also check if there's a color variable for this one? Thanks!

File: app/src/main/res/xml/customization_preferences.xml
Comment (gabrielluong) [inline app/src/main/res/xml/customization_preferences.xml:10]: I think these dividers were just added in https://github.com/mozilla-mobile/fenix/pull/15917. I am not quite sure we should be removing them.

**Review by gabrielluong — CHANGES_REQUESTED** [review]
@amkcpu Thanks for the contribution, I think the code looks good and is almost there.

One thing that concerned me was that we are removing the divider in both the customization and search preference where we might not want to be.  Can you double check the requirements? I also pinged @brampitoyo to do a UX review just in case. I know in https://github.com/mozilla-mobile/fenix/pull/15917 we just added these dividers working with @apbitner so it would be good to make sure we are align UX wise and are not undoing work.

File: app/src/main/res/values/colors.xml
Comment (amkcpu) [reply app/src/main/res/values/colors.xml]: Just pushed a commit for that

File: app/src/main/res/xml/customization_preferences.xml
Comment (amkcpu) [reply app/src/main/res/xml/customization_preferences.xml:10]: I added `app:allowDividerAbove="true"` in `preference_cat_style`, I removed this here to avoid redundancy. The dividers are still there.

**Review by brampitoyo — APPROVED** [review]
@amkcpu Great work!

Will this styling be applied everywhere? My big question is the “Add-ons” section.

r+ looks good!

**Comment by amkcpu** [comment]
@brampitoyo Thanks!

> Will this styling be applied everywhere? My big question is the “Add-ons” section.

I did not do the add-ons section in this PR because that needs AC work. There is an issue for that: #15919.
Apart from that, it's applied everywhere.

**Comment by gabrielluong** [comment]
Sorry for the delay, I will get to this tomorrow!

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
Rebased and landed in https://github.com/mozilla-mobile/fenix/pull/16332. Thank you again @amkcpu!

**Pull request closed without merging**
